### ISSUE TITLE
Support Console.beep on Linux (Fedora/RHEL)

### ISSUE THREAD

**Issue by Petermarcu** [dotnet/runtime#24366]
@chiddekel commented on [Tue Dec 05 2017](https://github.com/dotnet/core/issues/1130)

Run on OS: Fedora 27  
Kernel Linux:  4.13.16-302.fc27.x86_64  

Step zero: `yum install beep`  
First, run` sudo modprobe pcspkr` and then beep should work.
  
Additional uncomment the /etc/modprobe.d/beep.conf line at:
`# alias platform:pcspkr pcspkr`
to
`alias platform:pcspkr pcspkr`
For automatic load pcspkr module  

 Run application beep e.g -  `/usr/bin/beep  -f 2000 -l 1000 ` 
 The beep in this case equal to n = 2000 Hz and duration 1000 ms

This is tiny, small C source includes, of beep application. 
 https://goo.gl/E3kfPy  

The portion of this code can be re-write to C# and use in Console.beep implementation for Linux platform.

**Comment by webczat** [comment]
Just sharing what I know about this beeping/linux terminal stuff, in case you find it useful:
* This would likely make sense to work only when the user is at the console, either gui or text mode, is that right? Or should it work always even if you would connect through ssh and issue Beep...
* Generally you can beep by issuing some ioctls on a virtual terminal. In case of ssh or serial port there is no current vt, in case of text mode there is, in case of graphical mode there is, but only the xserver/wayland compositor is really connected to it.
* /dev/tty won't work for graphical sessions or remote, /dev/ttyn would in case of gui but not sure how to check which to open (the correct one has ownership set to current user), /dev/tty0 is the currently active vt but it is root only, and /dev/console does not even have to be the vt.
* The pc speaker also registers as input device but I am not really sure how much it matters. Although generally you may be able to get access to these from non remote without root. But finding the correct one means probably checking in sysfs or udev. I am not sure what is or is not acceptable.
In summary, to me it seems at least tricky.

**Comment by richlander** [comment]
Just saw this issue. I don't think I saw it originally. I am closing this given how old this issue is, and the fact that the primary code being considered is GPL.

Anyone should feel to open a new issue if they have a new proposal for tackling this topic.
